src/thread.cpp

```diff
@@ -92,10 +93,32 @@ void zmq::thread_t::setSchedulingParameters (
     LIBZMQ_UNUSED (affinity_cpus_);
 }
 
+
 void zmq::thread_t::setThreadName (const char *name_)
 {
-    // not implemented
-    LIBZMQ_UNUSED (name_);
+    if (!name_)
+        return;
+
+    struct {
+        DWORD _type;
+        LPCSTR _name;
+        DWORD _thread_id;
+        DWORD _flags;
+    } thread_info;
+
+	thread_info._type = 0x1000;
```

Comment:
The indentation seems to be wrong here.

Reply:
Sorry about that, I fixed this on a badly configured editor..I'll open a new PR